framework/src/timeintegrators/LStableDirk2.C

```diff
@@ -0,0 +1,151 @@
+/****************************************************************/
+/*               DO NOT MODIFY THIS HEADER                      */
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*           (c) 2010 Battelle Energy Alliance, LLC             */
+/*                   ALL RIGHTS RESERVED                        */
+/*                                                              */
+/*          Prepared by Battelle Energy Alliance, LLC           */
+/*            Under Contract No. DE-AC07-05ID14517              */
+/*            With the U. S. Department of Energy               */
+/*                                                              */
+/*            See COPYRIGHT for full restrictions               */
+/****************************************************************/
+
+#include "LStableDirk2.h"
+#include "NonlinearSystem.h"
+#include "FEProblem.h"
+#include "PetscSupport.h"
+
+template<>
+InputParameters validParams<LStableDirk2>()
+{
+  InputParameters params = validParams<TimeIntegrator>();
+  return params;
+}
+
+
+LStableDirk2::LStableDirk2(const InputParameters & parameters) :
+    TimeIntegrator(parameters),
+    _stage(1),
+    _residual_stage1(_nl.addVector("residual_stage1", false, GHOSTED)),
+    _residual_stage2(_nl.addVector("residual_stage2", false, GHOSTED)),
+    _solution_start(_sys.solutionOld()),
+    _alpha(1. - 0.5*std::sqrt(2))
+{
+}
+
+
+
+LStableDirk2::~LStableDirk2()
+{
+}
+
+
+
+void
+LStableDirk2::computeTimeDerivatives()
+{
+  // We are multiplying by the method coefficients in postStep(), so
+  // the time derivatives are of the same form at every stage although
+  // the current solution varies depending on the stage.
+  _u_dot  = *_solution;
+  _u_dot -= _solution_old;
+  _u_dot *= 1. / _dt;
+  _u_dot.close();
+  _du_dot_du = 1. / _dt;
+}
+
+
+
+void
+LStableDirk2::solve()
+{
+  // Time at end of step
+  Real time_new = _fe_problem.time();
+
+  // Time at beginning of step
+  Real time_old = _fe_problem.timeOld();
+
+  // Time at stage 1
+  Real time_stage1 = time_old + _alpha*_dt;
+
+  // Solution at beginning of time step; store it because it is needed in update step
+  _solution_start = _solution_old;
+
+  // Compute first stage
+  _console << "LStableDirk2: 1. stage" << std::endl;
```

Comment:
std::endl? I'm not completely opposed but we generally try to avoid these. We are down to only about 6 throughout the framework. If we really want it here that's ok, but if it's not completely necessary, `'\n'` is preferred.